subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestOptionsIntegrationSpec.groovy

```diff
@@ -405,6 +407,179 @@ public class SomeIntegTestClass {
     }
     // endregion stand-alone test tasks
 
+    @Issue("https://github.com/gradle/gradle/issues/24331")
+    def "options can be set with framework before java plugin is applied"() {
```

Comment:
💭 Is this really "before"?  The lazy configuration block won't run prior to the plugin application.

Reply:
It registers a task rule that runs "before" anything else.  So we're really talking about the order of the rules, not the order of the configuration, although it just so happens that the order of the configuration in the script corresponds to the order of the rules that get fired on the task object.  Happy to change this if you think there is a better way to express this distinction.